### PR TITLE
Create github action to check consistency of requirements.txt and notebook_requirements.txt

### PR DESCRIPTION
**Pull request by kanderso-nrel**
- [x] Code changes are covered by tests
- [x] Code changes have been evaluated for compatibility/integration with TrendAnalysis
- [x] New functions added to `__init__.py`
- [x] API.rst is up to date, along with other sphinx docs pages
- [x] Example notebooks are rerun and differences in results scrutinized
- [x] Updated changelog

@mdeceglie suggested adding a job to verify the consistency of `requirements.txt` and `notebook_requirements.txt` to make it easier to review PRs from dependabot.  Note that consistency of `requirements.txt` is already checked by the pytest jobs, but currently no CI job uses `notebook_requirements.txt`.  Also note that dependabot does suggest updates to `notebook_requirements.txt`, not just `requirements.txt`, e.g. #288.  Maybe this job will become redundant if we ever get around to #270. 

I tested some inconsistent environments on my fork to see what failures look like; here are the corresponding jobs:
- inconsistent 1: https://github.com/kanderso-nrel/rdtools/runs/3737515003?check_suite_focus=true
- inconsistent 2: https://github.com/kanderso-nrel/rdtools/runs/3737560421?check_suite_focus=true
- consistent: https://github.com/kanderso-nrel/rdtools/runs/3737621395?check_suite_focus=true

Note that `inconsistent 1` requested a version of numpy without wheels for py 3.7, so it tried to compile it from source and it was that compilation that failed, not a consistency error, so it's not really relevant.  But `inconsistent 2` shows an actual consistency failure.

### CODE REVIEW

File: .github/workflows/requirements.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+name: requirements
+
+on: [pull_request, push]
+
+jobs:
+  requirements:
+
+    runs-on: ubuntu-latest
```
Comment (kanderso-nrel) [inline .github/workflows/requirements.yaml:8]: Should we test other OSs?  I guess it depends on whether we are checking that the environment can be installed easily, or checking whether the package versions are compatible (which is a weaker requirement than the package versions can be installed easily on all OSs).

File: .github/workflows/requirements.yaml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.7
```
Comment (kanderso-nrel) [inline .github/workflows/requirements.yaml:15]: Same question as above, but for different python versions.  I don't think we claim that the notebooks support anything other than 3.7

File: .github/workflows/requirements.yaml (hunk 1 above)
Comment (mdeceglie) [reply .github/workflows/requirements.yaml:8]: Although I'm a little worried we will open a can of worms, it does sound useful to check on windows and osx. Can we give it a try?

File: .github/workflows/requirements.yaml (hunk 2 above)
Comment (mdeceglie) [reply .github/workflows/requirements.yaml:15]: I'm fine with sticking with just 3.7

File: .github/workflows/requirements.yaml (hunk 1 above)
Comment (mdeceglie) [reply .github/workflows/requirements.yaml:8]: Or maybe just add the other OSs to the pytest workflow?

**Comment by mdeceglie** [comment]
This looks good, but i'm imagining cutting 2.1.1, so we should probably update the changelog about the xgboost requirement

**Comment by kanderso-nrel** [comment]
Given this is just a maintenance PR, do we need to cut a release?  I imagined the xgboost requirement change to be a short-term change that gets reverted before we're ready to cut another release.  If that's the case then I don't see a point in a changelog entry.

**Comment by mdeceglie** [comment]
I think we should do a patch, an updated `setup.py` has limited utility sitting on github, we should get it out in pypi

**Comment by kanderso-nrel** [comment]
Oh right, if it's broken for us, it's broken for everyone.  Forgot about that part.  Changelog added!

**Review by mdeceglie — APPROVED** [review]
Looks good to me, thanks @kanderso-nrel !

**Pull request merged**